diff --git a/node_modules/prismarine-chunk/node_modules/minecraft-data/data.js b/node_modules/prismarine-chunk/node_modules/minecraft-data/data.js
index e94aa62..2ef0fd2 100644
--- a/node_modules/prismarine-chunk/node_modules/minecraft-data/data.js
+++ b/node_modules/prismarine-chunk/node_modules/minecraft-data/data.js
@@ -1,5 +1,6 @@
-module.exports =
-{
+const __dirname = new URL('.', import.meta.url).pathname;
+
+export default {
   'pc': {
     '0.30c': {
       get blocks () { return require("./minecraft-data/data/pc/0.30c/blocks.json") },
diff --git a/node_modules/prismarine-chunk/node_modules/minecraft-data/index.js b/node_modules/prismarine-chunk/node_modules/minecraft-data/index.js
index 70cb787..8a8a8bc 100644
--- a/node_modules/prismarine-chunk/node_modules/minecraft-data/index.js
+++ b/node_modules/prismarine-chunk/node_modules/minecraft-data/index.js
@@ -130,4 +130,4 @@ const schemas = {
 }
 module.exports.schemas = schemas
 
-const data = require('./data.js')
+const data = require('./data.js').default
